### DISCUSSION TITLE
How to remove comments?

### DISCUSSION THREAD

**Discussion by jeffrson** [swc-project/swc#4446] [Q&A]
Hi,

there is jsc.preserveAllComments - but what does it mean? What is removed, what isn't?

And how can I remove all comments without minifying? `minify:{}` removes comments, but minifies if/else (for example).

Thanks!

**Comment by kdy1** [comment]
`minify: { compress: false, mangle: false }` will work I guess?

`jsc.minify.formt` can be used to control comment removing behavior

**Comment by jeffrson** [comment]
Thank you for this suggestion! This seems to behave the same as `minify: {}`

It changes code like this:
```
  return fs
        .readFile(configFilePath, 'utf8')
        .then((configRaw) => {
            if (configRaw.charCodeAt(0) === 0xfeff) {
                configRaw = configRaw.slice(1)
            }

            return parse(configRaw) as IConfiguration
        })
        .catch(() => undefined)
```

into

```
    return fs.readFile(configFilePath, 'utf8').then((configRaw)=>{
        if (configRaw.charCodeAt(0) === 0xfeff) {
            configRaw = configRaw.slice(1);
        }
        return parse(configRaw);
    }).catch(()=>undefined
    );
```
while tsc leaves it essentially like it is. The same for (this is a template, don't judge code usefulness ;-)):

```
  if (os.platform() === 'win32') {
        //
    } else if (os.platform() === 'linux') {
        //
    } else if (os.platform() === 'darwin') {
        //
    } else {
        //
    }
```

becomes

```
if (os.platform() === 'win32') {} else if (os.platform() === 'linux') {} else if (os.platform() === 'darwin') {} else {}
```

It would make it easier to compare with tsc if it looked the same.
